### ISSUE TITLE
Silent Hill 4 (Silent Hill 4: The Room) [4B4E001F]

### ISSUE THREAD

**Issue by hotzenplotz3000** [Cxbx-Reloaded/game-compatibility#336]
[Silent Hill 4: The Room](https://en.wikipedia.org/wiki/Silent_Hill_4:_The_Room)

### Description

Game is booting, but only shows a black screen
You can hear the intro playing and the menu sounds.
Sound is cracking and after starting the game it crashes with XID_fCloseDevice+0x3e5cc10

### Dumps 

[Xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1290655/Xbe.txt)
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1290656/KrnlDebug.txt)
[hlecache.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1290657/hlecache.txt)

### Status
**Previous behavior:** -

**Current behavior:** booting

### Screenshots 
![capture20170910172901661](https://user-images.githubusercontent.com/30478365/30250474-4e3de2d6-964f-11e7-912a-2e5c4652f975.png)

### Additional Information

tested with build 8cf6883e

**Comment by hotzenplotz3000** [comment]
Checked current state with build 984e2bd5
Game crashes when it start playing the Intro:
```
---------------------------
Cxbx-Reloaded
---------------------------
Received Exception Code 0xC0000005 @ EIP := 0x0405477C

  Press "OK" to terminate emulation.
  Press "Cancel" to debug.
---------------------------
OK   Abbrechen   
---------------------------
```

Right before this exception, this warning is shown in the log:
> [0x28E0] WARN: Out of physical memory!

[KrnlDebug_984e2bd5.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1596372/KrnlDebug_984e2bd5.txt)

**Comment by ergo720** [comment]
That message comes from the manager and indicates it has exhausted the main memory.

**Comment by PatrickvL** [comment]
Running out of physical memory so early in, is a clear sign that our HLE code is leaking memory like crazy

**Comment by hotzenplotz3000** [comment]
for the sake of completeness, a log from the debug build:
[KrnlDebug_dbg.zip](https://github.com/Cxbx-Reloaded/game-compatibility/files/1596438/KrnlDebug_dbg.zip)

**Comment by ergo720** [comment]
@PatrickvL Indeed, the last attempted allocation comes from D3DTexture_GetSurfaceLevel2 which calls EmuNewD3DSurface and then g_VMManager.AllocateZeroed

**Comment by Ale100000** [comment]
Tested on https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/commit/d03151cb5c99de010107e2f2eb14cfca088a6604
[Xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/2087997/Xbe.txt)
[KrnlDebugHLE.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/2087998/KrnlDebugHLE.txt)
[KrnlDebugLLE.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/2087999/KrnlDebugLLE.txt)
With HLE disc error,with LLE in-game with low fps and the emulation stops after some minutes or a CreateSoundBuffer error
![hl](https://user-images.githubusercontent.com/16784631/41203250-bd8a5854-6caa-11e8-856e-5cd67b3095f0.png)
LLE
![00](https://user-images.githubusercontent.com/16784631/41203323-ae5f5874-6cab-11e8-99a4-d9230a60d203.png)
![01](https://user-images.githubusercontent.com/16784631/41203256-d3672aa8-6caa-11e8-82b8-9c107a180792.png)
![02](https://user-images.githubusercontent.com/16784631/41203258-d449c4bc-6caa-11e8-8804-bd74ac34eab5.png)
![03](https://user-images.githubusercontent.com/16784631/41203261-d5443f6e-6caa-11e8-943e-47cbc73bded3.png)
![05](https://user-images.githubusercontent.com/16784631/41203371-367fdf1c-6cac-11e8-9dd0-82457f9d2c41.jpg)

**Comment by xdsid** [comment]
I'm new to this... where's the download link?

**Comment by mainzerbe** [comment]
Ingame now with HLE.

![silent](https://user-images.githubusercontent.com/66551412/99907954-000bc580-2ce0-11eb-883e-2c0c27612e7f.PNG)
